### ISSUE TITLE
New Features

### ISSUE THREAD

**Issue by Magnogen** [Magnogen/Elements#7]
I'd like to add a few more features, as three isn't that much lol. Any other ideas for this? Hmm

- [ ] Canvas
 
  A way to draw a bunch of stuff without having to do arcs for circles translate and rotate for just rotated things, and things like that. P5.js inspired.
- [ ] Range
 
  `0 ... 5`, `!0 ... 1`, etc, basically a shorter way to do conditions. Based on [ColourTode by TodePond](https://todepond.gitbook.io/colourtode/#range)
- [ ] Vector

  `add`, `sub`, `dot`, `cross`, `div`, `lerp`, stuff like that - but for vectors of arbitrary dimensions!
- [x] State
  
  Basically FSMs, a way to switch between states and stuff using an object that has objects or whatever.
  
  I have no idea how to implement this.
- [x] Thread

  A way to do multiple things in sort-of parallel. Good for generative art that can take a while haha. Pixel manipulation and lots of circles, maybe even simulations and such.
